src/plugins/intel_cpu/src/nodes/rnn.cpp

```diff
@@ -373,17 +373,22 @@ void RNN::configurePortDataTypes() {
     if (!is_cell)
         inDataTypes[sIdx] = memory::data_type::s32;
     inDataTypes[wIdx] = DnnlExtensionUtils::IEPrecisionToDataType(getOriginalInputPrecisionAtPort(wIdx));
-    inDataTypes[rIdx] = DnnlExtensionUtils::IEPrecisionToDataType(getOriginalInputPrecisionAtPort(wIdx));
+    inDataTypes[rIdx] = DnnlExtensionUtils::IEPrecisionToDataType(getOriginalInputPrecisionAtPort(rIdx));
 
     inDataTypes[bIdx] = memory::data_type::f32;
     if (haveAttention(cell_type))
         inDataTypes[aIdx] = DnnlExtensionUtils::IEPrecisionToDataType(getOriginalInputPrecisionAtPort(aIdx));
 
     if (!is_cell)
         outDataTypes[yIdx] = DnnlExtensionUtils::IEPrecisionToDataType(getOriginalOutputPrecisionAtPort(0));
+
     outDataTypes[hoIdx] = inDataTypes[hIdx]; // required by oneDNN. Output hidden state is a input hidden state for the next iteration
+
     if (haveCellState(cell_type))
         outDataTypes[coIdx] = inDataTypes[cIdx]; // required by oneDNN.
+
+    if (inDataTypes[xIdx] == memory::data_type::bf16)
```

Comment:
Data type requirements of oneDNN - https://oneapi-src.github.io/oneDNN/dev_guide_rnn.html#implementation-details

Reply:
Did not get it. Is there something wrong with the current change?

Reply:
Corrected to cover more potential use cases